lib/src/util.dart

```diff
@@ -72,3 +76,31 @@ bool hasHeader(Map<String, String> headers, String name) {
   }
   return false;
 }
+
+Map<String, List<String>> expandToHeadersAll(
+    Map<String, /* String | List<String> */ dynamic> headers) {
+  if (headers is Map<String, List<String>>) return headers;
+  if (headers == null || headers.isEmpty) return null;
+
+  return Map<String, List<String>>.fromEntries(headers.entries.map((e) {
+    return MapEntry<String, List<String>>(e.key, expandHeaderValue(e.value));
+  }));
+}
+
+List<String> expandHeaderValue(dynamic v) {
+  if (v is String) {
+    return [v];
+  } else if (v is List<String>) {
+    return v;
+  } else {
+    throw ArgumentError('Expected String or List<String>, got: `$v`.');
+  }
+}
```

Comment:
The issue @natebosch mentions looks to origin from here... 
```
Error thrown by handler.
Invalid argument(s): Expected String or List<String>, got: `null`.

package:shelf/src/response.dart 66:9                             new Response.ok
```
In the current release the following code works:
```dart
  test('allows null header', () {
    var response = Response.ok('hello, world', headers: {'x-hack': null});
    expect(response.readAsString(), completion(equals('hello, world')));
  });
```
I don't think that'll work on this branch as we'll throw an `ArgumentError` here.